mqttshark

```diff
@@ -351,20 +351,32 @@ def set_client_mapping(packet):
 
 
 def set_client_version(packet, ver):
-    src = "%s:%s" % (packet['layers']['ip']['ip_ip_src'], packet['layers']['tcp']['tcp_tcp_srcport'])
-    dst = "%s:%s" % (packet['layers']['ip']['ip_ip_dst'], packet['layers']['tcp']['tcp_tcp_dstport'])
+    src = "%s:%s" % (get_packet_layer_ip_peer(packet, 'src'), packet['layers']['tcp']['tcp_tcp_srcport'])
+    dst = "%s:%s" % (get_packet_layer_ip_peer(packet, 'dst'), packet['layers']['tcp']['tcp_tcp_dstport'])
     client_versions[src] = ver
     client_versions[dst] = ver
 
+
 def get_client_version(packet):
-    src = "%s:%s" % (packet['layers']['ip']['ip_ip_src'], packet['layers']['tcp']['tcp_tcp_srcport'])
+    src = "%s:%s" % (get_packet_layer_ip_peer(packet, 'src'), packet['layers']['tcp']['tcp_tcp_srcport'])
     try:
         return client_versions[src]
     except KeyError:
         return 4
 
+
+def get_packet_layer_ip_peer(packet, kind):
+    assert kind in ['src', 'dst'], f'Peer kind invalid: {kind}'
+    if 'ip' in packet['layers']:
+        return packet['layers']['ip'][f'ip_ip_{kind}']
+    elif 'ipv6' in packet['layers']:
+        return packet['layers']['ipv6'][f'ipv6_ipv6_{kind}']
+    else:
+        raise ValueError(f'No "ip" or "ipv6" in packet layer')
```

Comment:
Please advise on "naming things", if you see a chance to improve on that detail.

For example, to make it a bit shorter, and reflect its purpose with a `decode_` prefix:
```python
def decode_ip_peer(packet, peer_type)
```